### ISSUE TITLE
nonfree code

### ISSUE THREAD

**Issue by nschloe** [xiaoyeli/superlu#9]
The file `License.txt` claims that SuperLU is distributed under BSD, but that doesn't seem to entirely be the case. I'm looking at [SRC/mc64ad.c](https://github.com/xiaoyeli/superlu/blob/master/SRC/mc64ad.c). One reads there:

> The user shall not pass this code directly to a third party without the
> express prior consent of the authors.  Users wanting to licence their
> own copy of these routines should send email to hsl@stfc.ac.uk.

Perhaps this condition is already violated by putting SuperLU on GitHub, and in any case is not something anyone would like to have to deal with.

Is the corresponding code still in use at all?

**Comment by xiaoyeli** [comment]
This is documented in the Users' Guide.  I am updating the License.txt file to include this. It is used as an option in the ILU driver routine, and is not essential.  The commercial users can easily comment out this use path, hence avoid the license issue.

**Comment by nschloe** [comment]
>  I am updating the License.txt file to include this.

Thanks.

> The commercial users 

It wasn't the commercial users, but -- quite the contrary --, the Debian folks which brought the issue to my attention. They are are traditionally more aware of licensing issues there. In fact, the packaging team has to go great lengths to explicitly _remove_ the offending code from the source to make sure that they don't do anything illegal, namely distributing the code. This licensing issue isn't removed for commercial users by commenting out a path either.

I'm not sure what the best thing to do is here. Could there be a "free" edition of SuperLU perhaps where everything is licensed under a BSD-3, and a "nonfree" ("pro"/whatever) edition with nonfree code included?

**Comment by xiaoyeli** [comment]
That may be too much work.  Eventually we will replace MC64 by our new implementation, but right now this is low priority.

**Comment by nschloe** [comment]
> That may be too much work

I agree. Let's keep this issue open though until MC64 has been replaced. Very much looking forward to this!

**Comment by nschloe** [comment]
Btw there is more nonfree code at

 * MATLAB/spypart.m
